### PR TITLE
Radiocommunications in ContactDetails

### PR DESCRIPTION
**Pull request by rik**

### CODE REVIEW

File: s127/models/information_type.py [hunk 1]
```diff
@@ -298,6 +298,34 @@ class CategoryOfCommPref(models.TextChoices):
     # fmt: on
 
 
+# https://github.com/betagouv/SPPNautInterface/issues/244
+class FrequencyPair(s100.models.GenericComplexAttributeType):
```
Comment (rik) [inline s127/models/information_type.py:302]: Do we want to keep this? I have a vague recollection that SHOM told us it wouldn't be used.

File: s127/models/information_type.py [hunk 2]
```diff
@@ -359,6 +388,40 @@ class ContactDetails(s100.models.InformationType):
 
     information = GenericRelation(s100.models.Information)
 
+    def __str__(self):
```
Comment (rik) [inline s127/models/information_type.py:391]: We need tests. That will help us review what we want to display.

File: s127/models/information_type.py [hunk 3]
```diff
@@ -426,6 +489,69 @@ class Meta:
...
+    contact_instructions = models.TextField(
+        blank=True,
+        null=True,
+        help_text="Supplemental instructions on how or when to contact the individual, organisation, or service",
+    )
+
+    # Complex attribute used by multiple classes
+    frequency_pair = GenericRelation(FrequencyPair)
+
+    def __str__(self):
```
Comment (rik) [inline s127/models/information_type.py:536]: We need tests. That will help us review what we want to display.

File: s127/models/information_type.py [hunk 4]
```diff
@@ -484,5 +610,23 @@ class TelecommunicationService(s100.models.CodeList):
         help_text="The name of provider or type of carrier for a telecommunications service",
     )
 
+    def __str__(self):
```
Comment (rik) [inline s127/models/information_type.py:613]: We need tests. That will help us review what we want to display.

File: s127/models/information_type.py [hunk 5]
```diff
@@ -426,6 +489,69 @@ class Meta:
         verbose_name_plural = "Contact Addresses"
 
 
+class Radiocommunications(s100.models.ComplexAttributeType):
+    """
+    Detailed radiocommunications description with channels, frequencies, preferences and time schedules.
+
+    https://github.com/betagouv/SPPNautInterface/issues/273
+    https://github.com/betagouv/SPPNautInterface/issues/240
+    https://github.com/betagouv/SPPNautInterface/issues/241
```
Comment (rik) [inline s127/models/information_type.py:498]: Note to self: Review the issues.

Also, we tend to put those issues in comments, not docstrings.

File: s127/models/information_type.py (hunk 3 above)
Comment (rik) [reply s127/models/information_type.py:479]: We've removed the STR

File: s127/models/information_type.py (hunk 4 above)
Comment (rik) [reply s127/models/information_type.py:556]: Done.

**Review by JulieMoncorger — APPROVED** [review]

**Pull request merged**